common/web/gesture-recognizer/src/engine/headless/inputEngineBase.ts

```diff
@@ -42,15 +102,26 @@ export abstract class InputEngineBase<HoveredItemType, StateToken = any> extends
    * @returns
    */
   protected getConfigForId(identifier: number) {
+    // protected - so, used internally only within the input engines.
+    // `getTouchpointWithId` will perform the internal -> external ID mapping.
     return this.getTouchpointWithId(identifier).currentRecognizerConfig;
   }
 
   protected getStateTokenForId(identifier: number) {
+    // protected - so, used internally only within the input engines.
+    // `getTouchpointWithId` will perform the internal -> external ID mapping.
     return this.getTouchpointWithId(identifier).stateToken ?? null;
   }
 
-  public dropTouchpointWithId(identifier: number) {
-    this._activeTouchpoints = this._activeTouchpoints.filter((point) => point.rawIdentifier != identifier);
+  protected dropTouchpoint(point: GestureSource<HoveredItemType>) {
+    const id = point.rawIdentifier;
+
+    this._activeTouchpoints = this._activeTouchpoints.filter((pt) => point != pt);
+    for(let key in this.identifierMap) {
```

Comment:
`for(.. in ..)` is not [very safe](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#iterating_over_own_properties_only). It's also slow.

A safer alternative:

```suggestion
    for(const key of Object.keys(this.identifierMap)) {
```